src/clients/TokenClient.ts

```diff
@@ -203,54 +209,53 @@ export class TokenClient {
     this.logger.debug({ at: "TokenBalanceClient", message: "TokenBalance client updated!", balanceData });
   }
 
-  async fetchTokenData(spokePoolClient: SpokePoolClient): Promise<{
-    tokenData: Record<string, { balance: BigNumber; allowance: BigNumber }>;
-    chainId: number;
-  }> {
-    const tokens = spokePoolClient
-      .getAllOriginTokens()
-      .map((address) => new Contract(address, ERC20.abi, spokePoolClient.spokePool.signer));
+  async fetchTokenData(
+    chainId: number,
+    hubPoolTokens: string[],
+    signer: Signer
+  ): Promise<Record<string, { balance: BigNumber; allowance: BigNumber }>> {
+    const tokens = hubPoolTokens
+      .map((address) => {
+        try {
+          const spokePoolToken = this.hubPoolClient.getL2TokenForL1TokenAtBlock(address, chainId);
+          return new Contract(spokePoolToken, ERC20.abi, signer);
```

Comment:
It may be worth using the ERC20 class directly - feel free to ignore
```suggestion
          return ERC20.connect(spokePoolToken, signer)
```

Reply:
This is nice - I'll propose it for a separate PR since this one doesn't touch the TokenClient anymore.